pkg/asset/machines/azure/azuremachines.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	if len(mpool.Zones) == 0 {
+		// if no azs are given we set to []string{""} for convenience over later operations.
+		// It means no-zoned for the machine API
+		mpool.Zones = []string{""}
+	}
+	tags, err := CapzTagsFromUserTags(clusterID, userTags)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create machineapi.TagSpecifications from UserTags: %w", err)
+	}
+
+	var image capz.Image
+	osImage := mpool.OSImage
+	if osImage.Publisher != "" {
+		image = capz.Image{
+			Marketplace: &capz.AzureMarketplaceImage{
+				ImagePlan: capz.ImagePlan{
+					Publisher: osImage.Publisher,
+					Offer:     osImage.Offer,
+					SKU:       osImage.SKU,
+				},
+				Version: osImage.Version,
+			},
+		}
+	} else if useImageGallery {
+		// image gallery names cannot have dashes
+		galleryName := strings.Replace(clusterID, "-", "_", -1)
+		id := clusterID
+		if hyperVGen == "V2" {
+			id += "-gen2"
+		}
+		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/galleries/gallery_%s/images/%s/versions/latest", resourceGroup, galleryName, id)
+		image.ID = &imageID
+	} else {
+		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/images/%s", resourceGroup, clusterID)
+		if hyperVGen == "V2" && platform.CloudName != azure.StackCloud {
+			imageID += "-gen2"
+		}
+		image.ID = &imageID
+	}
+
+	osDisk := capz.OSDisk{
+		OSType:     "Linux",
+		DiskSizeGB: &mpool.DiskSizeGB,
+		ManagedDisk: &capz.ManagedDiskParameters{
+			StorageAccountType: mpool.DiskType,
+		},
+	}
+	ultrassd := mpool.UltraSSDCapability == "Enabled"
+	additionalCapabilities := capz.AdditionalCapabilities{
+		UltraSSDEnabled: &ultrassd,
+	}
+
+	diagnostics := capz.Diagnostics{
+		Boot: nil,
+	}
+
+	machineProfile := generateSecurityProfile(mpool)
+	securityProfile := capz.SecurityProfile{
+		EncryptionAtHost: machineProfile.EncryptionAtHost,
+		SecurityType:     capz.SecurityTypes(machineProfile.Settings.SecurityType),
+	}
+	if machineProfile.Settings.ConfidentialVM != nil {
+		securityProfile.UefiSettings = &capz.UefiSettings{
+			VTpmEnabled:       ptr.To(machineProfile.Settings.ConfidentialVM.UEFISettings.VirtualizedTrustedPlatformModule == "Enabled"),
+			SecureBootEnabled: ptr.To(machineProfile.Settings.ConfidentialVM.UEFISettings.SecureBoot == "Enabled"),
+		}
+	} else if machineProfile.Settings.TrustedLaunch != nil {
+		securityProfile.UefiSettings = &capz.UefiSettings{
+			VTpmEnabled:       ptr.To(machineProfile.Settings.TrustedLaunch.UEFISettings.VirtualizedTrustedPlatformModule == "Enabled"),
+			SecureBootEnabled: ptr.To(machineProfile.Settings.TrustedLaunch.UEFISettings.SecureBoot == "Enabled"),
+		}
+	}
+
+	providerID := fmt.Sprintf("%s-identity", clusterID)
+	var result []*asset.RuntimeFile
+	for idx := int64(0); idx < total; idx++ {
+		zone := mpool.Zones[int(idx)%len(mpool.Zones)]
+
+		networkInterfaces := []capz.NetworkInterface{{
+			SubnetName:            subnet,
+			AcceleratedNetworking: additionalCapabilities.UltraSSDEnabled,
```

Comment:
Is `UltraSSDEnabled` in the correct place here? Seems that one is related to disks, rather than network interfaces?

(not super familiar with Azure specifically, was going by https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.management.compute.models.additionalcapabilities.ultrassdenabled?view=azure-dotnet-legacy)